components/client/src/main/java/com/hotels/styx/client/StyxBackendServiceClient.java

```diff
@@ -140,22 +136,33 @@ private static boolean isHeadRequest(LiveHttpRequest request) {
         Optional<RemoteHost> remoteHost = selectOrigin(request);
         if (remoteHost.isPresent()) {
             RemoteHost host = remoteHost.get();
+
+            LiveHttpRequest withHostHeaderUpdated = request;
```

Comment:
This logic could be extracted as a separate method.
If we do convert this class to Kotlin, the method would be most intuitive as a (private) extension method.